src/kademlia/dht_tracker.cpp

```diff
@@ -294,26 +251,25 @@ namespace libtorrent { namespace dht
 		, boost::function<void(item const&, bool)> cb
 		, std::string salt)
 	{
-		m_dht.get_item(key, salt, boost::bind(&get_mutable_item_callback, _1, _2, cb));
+		m_dht.get_item(key, salt, cb);
 	}
 
 	void dht_tracker::put_item(entry data
-		, boost::function<void()> cb)
+		, boost::function<void(int)> cb)
 	{
 		std::string flat_data;
 		bencode(std::back_inserter(flat_data), data);
 		sha1_hash target = item_target_id(
 			std::pair<char const*, int>(flat_data.c_str(), flat_data.size()));
 
-		m_dht.get_item(target, boost::bind(&put_immutable_item_callback
-			, _1, cb, data));
+		m_dht.put_item(target, data, cb);
 	}
 
 	void dht_tracker::put_item(char const* key
-		, boost::function<void(item&)> cb, std::string salt)
+		, boost::function<void(item&, int)> cb
```

Comment:
granted, before this interface reaches the user, it should be simpler and probably separate get and put. In the intermediate layers the interfaces should probably be as narrow as possible (that mostly applies to the user-facing interface as well). Now, I think dht_tracker is probably entirely redundant too, and just cause overhead in passing calls back and forth.